frontend/src/components/navigation/AppNavBar.tsx

```diff
@@ -62,11 +58,11 @@ function AppNavBar() {
       <Navbar.Toggle aria-controls="collapse" className="navbar-dark" />
       <Nav>
         <Navbar.Collapse className="links">
-          {canAdmin}
-          {canEditProperties}
-          {canPropertyView}
-          {canDisposeRequest}
-          {canDisposeApprove}
+          {AdminDropdown}
```

Comment:
As you are creating sub-conponents it would be a bit cleaner to use the JSX syntax here - `<AdminDropdown/>`

Reply:
Do we really want to make these sub-components?  They're essentially just logic to determine which nav elements are displayed.

To make them sub-components I'd be creating separate files in the components correct?

Reply:
@Fosol see here: https://reactjs.org/docs/introducing-jsx.html

Reply:
Another way is to turn the above into "render" functions. That's another React pattern I've seen around. E.g.

```jsx
const renderAdminDropdown = () => {
  if (you_have_permission) {
    return <NavLink ... />
  } else {
    return null;
  }
}
``` 
And then in the main JSX body use:

```diff
- {AdminDropdown}
+ { renderAdminDropdown() } <-- note the function call
```
Both approaches are acceptable - I personally like the JSX components better

Reply:
What's the best way to create these sub-components?  If I do the following,
```
  function SubmitPropertyNav() {
    return keycloak.hasClaim(Claims.PROPERTY_ADD) ? (
      <NavDropdown.Item onClick={() => history.push('/submitProperty')}>
        Submit Property
      </NavDropdown.Item>
    ) : null;
  }
```
Then it needs to belong in the parent `AppNavBar()` component, as it needs access to `keycloak` and `history` objects.

If add them to the function as arguments it apparently isn't ideal.  The recommendation is to add a `props` argument to the sub-components.  But then I think I still need to manually pass them.